### PR TITLE
Added Collation support

### PR DESCRIPTION
**Pull request by rozza**
SCALA-249

### CODE REVIEW

File: driver/src/main/scala/org/mongodb/scala/model/Collation.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+import com.mongodb.client.model.{ Collation => JCollation }
+
+/**
+ * The options regarding collation support in MongoDB 3.4+
+ *
+ * @note Requires MongoDB 3.4 or greater
+ * @since 1.2
+ */
+object Collation {
```
Comment (jyemin) [inline driver/src/main/scala/org/mongodb/scala/model/Collation.scala:27]: I see CollationAlternate and CollationMaxVariable, but am unsure how they are used.  Is there a test of how one builds a Scala driver Collation object with them?

Also, why are CollationAlternate and CollationMaxVariable wrapped but not CollationStrength or CollationCaseFirst?

Comment (rozza) [reply driver/src/main/scala/org/mongodb/scala/model/Collation.scala:27]: No need for a test - they are just mirrors to the enum values and are already tested to ensure they mirror and return the java values.

Not sure why CollationStrength and CollationCaseFirst didn't cause flag up in the API mirror tests. Will check that out.

**Comment by rozza** [comment]
Found out how to capture all Enums using reflection - so shouldn't miss anymore in the future.

File: driver/src/test/scala/org/mongodb/scala/model/CollationSpec.scala [hunk 2]
```diff
@@ -38,4 +45,32 @@ class CollationSpec extends FlatSpec with Matchers {
     Collation.builder().getClass should equal(classOf[com.mongodb.client.model.Collation.Builder])
   }
 
+  it should "produce the same collation value when using the Scala helpers" in {
```
Comment (jyemin) [inline driver/src/test/scala/org/mongodb/scala/model/CollationSpec.scala:48]: I like the test, even if it's redundant.

**Comment by jyemin** [comment]
LGTM

**Pull request closed without merging**